pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/ModeAggregationFunction.java

```diff
@@ -263,11 +264,15 @@ public void aggregate(int length, AggregationResultHolder aggregationResultHolde
     // For dictionary-encoded expression, store dictionary ids into the dictId map
     Dictionary dictionary = blockValSet.getDictionary();
     if (dictionary != null) {
-      int[] dictIds = blockValSet.getDictionaryIdsSV();
+
       Int2IntOpenHashMap dictIdValueMap = getDictIdCountMap(aggregationResultHolder, dictionary);
-      for (int i = 0; i < length; i++) {
-        dictIdValueMap.merge(dictIds[i], 1, Integer::sum);
-      }
+      int[] dids = blockValSet.getDictionaryIdsSV();
```

Comment:
(nit) Shall we keep the name as `dictIds` and `dictId`? That is used across all the engine code